cdap-client/src/main/java/co/cask/cdap/client/DatasetTypeClient.java

```diff
@@ -122,9 +168,27 @@ public boolean exists(Id.DatasetType type)
    * @throws UnauthenticatedException if the request is not authorized successfully in the gateway server
    * @throws TimeoutException if the dataset type was not yet existent before {@code timeout} milliseconds
    * @throws InterruptedException if interrupted while waiting
+   * @deprecated since 4.0.0. Use {@link #waitForExists(DatasetTypeId, long, TimeUnit)} instead.
    */
+  @Deprecated
   public void waitForExists(final Id.DatasetType type, long timeout, TimeUnit timeoutUnit)
     throws IOException, UnauthenticatedException, TimeoutException, InterruptedException {
+    waitForExists(type.toEntityId(), timeout, timeoutUnit);
+  }
+
+  /**
+   * Waits for a dataset type to exist.
+   *
+   * @param type the dataset type to check
+   * @param timeout time to wait before timing out
+   * @param timeoutUnit time unit of timeout
+   * @throws IOException if a network error occurred
+   * @throws UnauthenticatedException if the request is not authorized successfully in the gateway server
+   * @throws TimeoutException if the dataset type was not yet existent before {@code timeout} milliseconds
+   * @throws InterruptedException if interrupted while waiting
+   */
+  public void waitForExists(final DatasetTypeId type, long timeout, TimeUnit timeoutUnit)
```

Comment:
many of them are not used in cdap, not sure if they are used in other projects. I don't see what they are useful for though, especially since adding a type is a blocking call...